File: two_pointer/125. Valid Palindrome.cpp
#include <ctype.h>
#include <iostream>
#include <cstring>
#include <vector>

using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        
        int size = s.size();
        for(int i = 0, j = size - 1; i < j; ) {
            
            int ch1, ch2;
            ch1 = tolower(s[i]);
            ch2 = tolower(s[j]);
            
            if(ch1 < 97 && ch1 > 122)  i++;
            if(ch2 < 97 && ch2 > 122)  j--;
            if(ch1 == ch2) {
                i++;
                j--;
            } else  return false;
        }

        return true;
    }
};

File: two_pointer/125. Valid Palindrome.py

class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        i = 0
        j = len(s) - 1
        s = s.upper()
        
        while i < j:
            
            if not(s[i].isalpha() or s[i].isnumeric()):
                i = i + 1
                continue
            if not(s[j].isalpha() or s[j].isnumeric()):
                j = j - 1
                continue              
            if s[i] != s[j]:
                return False
            
            i = i + 1
            j = j - 1
        
        return True
            
            